server/api_admin.go

```diff
@@ -7,11 +7,138 @@ import (
 	"net/http"
 
 	"errors"
+	"fmt"
 
 	"github.com/gin-gonic/gin"
+	"github.com/mattermost/mattermost-plugin-ai/server/embeddings"
 	"github.com/mattermost/mattermost/server/public/model"
 )
 
+const defaultBatchSize = 100
+
+func (p *Plugin) handleReindexPosts(c *gin.Context) {
```

Comment:
Can this lead to an API timeout? Should we use something like a job here?

Reply:
Hmmm... yes you are correct. Plugins don't have access to the job server. I will think about how to address. this.